### ISSUE TITLE
Enhancement: [no-empty-interface] Allow extends in declaration

### ISSUE THREAD

**Issue by luxaritas** [typescript-eslint/typescript-eslint#6989]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-empty-interface/

### Description

It would be great to have a carve-out to no-empty-interface so that I can still fail single-extension in typical cases, but if single-extension is present in the context of declaration merging (ie, modifying an external interface), it does not fail.

### Fail

```typescript
interface CustomSessionData extends SessionData {}
```

### Pass

```typescript
import { SessionData } from 'express-session';
declare module 'express-session' {
  interface SessionData extends CustomSessionData {}
}
```

**Comment by luxaritas** [comment]
Actually, as I'm playing with this further, maybe this is actually covered by https://github.com/typescript-eslint/typescript-eslint/issues/2438? Not certain though, would appreciate guidance on whether this is a duplicate or not.

**Comment by bradzacher** [comment]
We don't look to add options that weaken the rule across the entire codebase just to stop it erroring in declaration files that use a different pattern to the rest of your codebase.

Instead, use disable comments. These have the added benefit of explicitly signalling intent and giving you a space to explain why you're breaking convention with the rest of the codebase so future contributors can quickly understand why the code doesn't follow conventions.

As for the fixer breaking declaration merging, that is covered by #2438 as you pointed out!

**Comment by luxaritas** [comment]
> just to stop it erroring in declaration files that use a different pattern

FWIW: In this case I'm writing this in-line with my code vs a separate declaration file (neither here nor there), but the reason I felt that this could use a separate configuration property is because declaration merging is _semantically different_. If there is an existing declaration for an interface, adding an `extends` is _not_ equivalent to the existence of the interface being extended from.
